### PR TITLE
Extends tests for ModuleStream.depends_on_stream()

### PR DESCRIPTION
**Pull request by athira-selvam**
The tests for depends_on_stream() and build_depends_on_stream() is modified to handle empty lists
Replicates the existing python tests, and newly added tests for depends_on_stream to C.
Fixes : https://github.com/fedora-modularity/libmodulemd/issues/192
related:  https://github.com/fedora-modularity/libmodulemd/issues/199

### CODE REVIEW

File: modulemd/v2/tests/ModulemdTests/modulestream.py [hunk 1]
```diff
@@ -1035,8 +1034,16 @@ def test_depends_on_stream(self):
                 stream.build_depends_on_stream(
                     'platform', 'f28'), False)
 
-            self.assertEqual(stream.depends_on_stream('base', 'f30'), False)
-            self.assertEqual(stream.depends_on_stream('base', 'f30'), False)
```
Comment (sgallagher) [inline modulemd/v2/tests/ModulemdTests/modulestream.py]: Why are you removing these two tests? Though, the second one should have been `stream.build_depends_on_stream()`, That's a bug in the original code here.

File: modulemd/v2/tests/ModulemdTests/modulestream.py [hunk 2]
```diff
@@ -682,7 +682,6 @@ def test_v2_yaml(self):
 
         assert 'rawhide' in stream.get_servicelevel_names()
         assert 'production' in stream.get_servicelevel_names()
-
```
Comment (sgallagher) [inline modulemd/v2/tests/ModulemdTests/modulestream.py]: Why delete this line?

File: modulemd/v2/tests/ModulemdTests/modulestream.py [hunk 3]
```diff
@@ -1035,8 +1034,16 @@ def test_depends_on_stream(self):
                 stream.build_depends_on_stream(
                     'platform', 'f28'), False)
 
-            self.assertEqual(stream.depends_on_stream('base', 'f30'), False)
-            self.assertEqual(stream.depends_on_stream('base', 'f30'), False)
+            self.assertEqual(
+                stream.depends_on_stream(
+                    'base', ''), False)
```
Comment (sgallagher) [inline modulemd/v2/tests/ModulemdTests/modulestream.py]: I think you misunderstood the requirement here. We don't need tests for a stream name being the empty string. We need a new requires/buildrequires in the v2 YAML document. Something like `perl` in:

```
  dependencies:
  - buildrequires:
      platform: [ f30 ]
      perl: []
    requires:
      platform: [ -f28 ]
      perl: []
```

This test is ONLY for ModuleStreamV2, since there is no way to represent this in the ModuleStreamV1 YAML.

**Comment by sgallagher** [comment]
Tracking this in #210 

Please do not file new PRs each time. Just update the current one by pushing to this branch.
